### PR TITLE
Added change to add Geopandas Dataframe to Datahub Space

### PR DESCRIPTION
**Pull request by omanges**
Signed-off-by: Omkar Mestry <omkar.mestry@here.com>

### CODE REVIEW

File: xyzspaces/spaces.py [hunk 1]
```diff
@@ -1228,3 +1211,25 @@ def add_features_geobuf(
             features_size=features_size,
             chunk_size=chunk_size,
         )
+
+    def add_features_geopandas(
+        self, data, features_size: int = 2000, chunk_size: int = 1
+    ):
+        """
+        To upload data from Geopandas Dataframe file to a space.
```
Comment (deeplook) [inline xyzspaces/spaces.py:1219]: Is that equivalent to "Add features from GeoPandas dataframe to a space."?

File: xyzspaces/spaces.py [hunk 2]
```diff
@@ -1228,3 +1211,25 @@ def add_features_geobuf(
...
+
+        :param data: Geopandas Dataframe to be uploaded
+        :param features_size: An int representing a number of features to upload at
```
Comment (deeplook) [inline xyzspaces/spaces.py:1222]: Skip the "int" as its visible in the signature, hence "The number of features..."

File: xyzspaces/spaces.py [hunk 3]
```diff
@@ -1228,3 +1211,25 @@ def add_features_geobuf(
             features_size=features_size,
             chunk_size=chunk_size,
         )
+
+    def add_features_geopandas(
+        self, data, features_size: int = 2000, chunk_size: int = 1
```
Comment (deeplook) [inline xyzspaces/spaces.py:1216]: Can we provide a type annotation for data?

File: xyzspaces/utils.py [hunk 4]
```diff
@@ -234,3 +235,33 @@ def divide_bbox(
         except Exception:
             logger.debug("The intersection geometry is incorrect")
     return final
+
+
+def flatten_geometry(data):
```
Comment (deeplook) [inline xyzspaces/utils.py:240]: Type annotation?

File: xyzspaces/utils.py [hunk 5]
```diff
@@ -234,3 +235,33 @@ def divide_bbox(
...
+    """
+    To flatten the geometries in the given Geopandas Dataframe.
```
Comment (deeplook) [inline xyzspaces/utils.py:242]: Skip the leading "To"... Explain what a flattened geometry is.

File: xyzspaces/utils.py [hunk 6]
```diff
@@ -234,3 +235,33 @@ def divide_bbox(
...
+
+    :param data: Geopandas Dataframe to be flatten
+    :return: flatten Geopandas Dataframe
```
Comment (deeplook) [inline xyzspaces/utils.py:245]: "Flattened (Flat?) GeoPandas dataframe"?

File: xyzspaces/spaces.py [hunk 7]
```diff
@@ -40,9 +40,10 @@
 import geobuf
 import geopandas as gpd
 from geojson import Feature, GeoJSON
+from geopandas import GeoDataFrame
```
Comment (sackh) [inline xyzspaces/spaces.py:43]: There is LGTM warning because of multiple imports of geopandas. maybe we can use the same import defined above?

Comment (omanges) [reply xyzspaces/spaces.py:43]: Can we?

**Review by sackh — APPROVED** [review]

**Pull request merged**